File: spec/files/Strings.m

#import "Strings.h"

NSString* LocalizedSingularString(SingularString singularString) {
    switch (singularString) {
        case SingularString_first_key:
            return NSLocalizedString(@"First key", @"");
        case SingularString_key_with_format:
            return NSLocalizedString(@"Key with format %ld", @"");
        case SingularString_key_without_description:
            return NSLocalizedString(@"Key without description", @"");
        default:
            return @"";
    }
}

File: spec/files/Strings.swift
import Foundation

struct Strings {

    enum Singular: String {
        case first_key = "First key"
        case key_with_format = "Key with format %ld"
        case key_without_description = "Key without description"

        var localizedString: String {
            return NSLocalizedString(rawValue, comment: "")
        }
    }
}
